receiver/apachereceiver/scraper.go

```diff
@@ -32,22 +34,49 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/apachereceiver/internal/metadata"
 )
 
+const (
+	EmitServerNameAsResourceAttribute = "receiver.apache.emitServerNameAsResourceAttribute"
+)
+
+var (
+	emitServerNameAsResourceAttribute = featuregate.Gate{
+		ID:      EmitServerNameAsResourceAttribute,
+		Enabled: false,
+		Description: "When enabled, the name of the server will be sent as an apache.server.name resource attribute " +
+			"instead of a metric-level server_name attribute.",
+	}
+)
+
+func init() {
+	featuregate.GetRegistry().MustRegister(emitServerNameAsResourceAttribute)
+}
+
 type apacheScraper struct {
 	settings   component.TelemetrySettings
 	cfg        *Config
 	httpClient *http.Client
 	mb         *metadata.MetricsBuilder
+
+	// Feature gates regarding resource attributes
+	emitMetricsWithServerNameAsResourceAttribute bool
 }
 
 func newApacheScraper(
 	settings component.ReceiverCreateSettings,
 	cfg *Config,
 ) *apacheScraper {
-	return &apacheScraper{
+	a := &apacheScraper{
 		settings: settings.TelemetrySettings,
 		cfg:      cfg,
 		mb:       metadata.NewMetricsBuilder(cfg.Metrics, settings.BuildInfo),
+		emitMetricsWithServerNameAsResourceAttribute: featuregate.GetRegistry().IsEnabled(EmitServerNameAsResourceAttribute),
 	}
+
+	if !a.emitMetricsWithServerNameAsResourceAttribute {
+		settings.Logger.Warn(fmt.Sprintf("Feature gate %s is not enabled. Please see the README.md file of apache receiver for more information.", EmitServerNameAsResourceAttribute))
```

Comment:
Maybe just put a link to README here?

Reply:
I added the link. However, I was not sure which branch/tag should we link to - I picked `main`, to avoid predicting if the next stable version will be `v0.63.0` or some other `v0.63.x` like in case of `v0.57`.